slides/week0/week0.org

```diff
@@ -0,0 +1,484 @@
+- How to contact me
+  - Slack: [[https://robojackets.slack.com/messages/@jasongibson/][@jasongibson]]
+  - Email: [[mailto:jgibson37@gatech.edu][jgibson37@gatech.edu]]
+
+* What Programming Is
+#+BEGIN_NOTES
+Talk for a minute or two about everything programming can do
+#+END_NOTES
+
+* What Programming Is Not
+#+BEGIN_NOTES
+This is meant to emphasize that they will not become a God programmer overnight
+and to dispell the misconception that they are going to be coding amazing things
+immediately.
+#+END_NOTES
+- Impossible
+- Easy
+
+* Programming Failures
+#+BEGIN_NOTES
+Talk for a minute or two about everything programming can do
+Orbiter was pounds instead of newtons
+Robocup forgot the flip the direction on the field and tried (unsuccessfully) to score on themselves
+Jason tried to delete a local directory usr and did sudo rm -rf /usr on a production server
+Ariane software testing did not acccount for increased speed and tried to store a number
+    that was not 16 bits in 1 bits and ended up in an explosion
+#+END_NOTES
+- Mars Climate Orbiter (1998)
+    - the code returned the wrong units causing a 655 million dollar explosion on Mars
+- RoboCup field disaster
+- deleting the /bin directory on a production server
+- Ariane s5 flight 501
+    - faster speed resulted in a 64 bit number being stored as 16 bits ~500 million failure
+
+* Codepad
+- Today we will be using codepad to write code in
+  - This is a remote interview tool
+- go to [[https://codepad.remoteinterview.io]]
+
+* Change to C++ The superior language
+#+ATTR_HTML: :width 50%
+[[file:https://i.imgur.com/IeHxwAM.png]]
+
+* Change to C++ The superior language
+#+ATTR_HTML: :width 50%
+[[file:https://i.imgur.com/S1PecNy.png]]
+
+* Hello World!
+- Add *World!* to your code
+#+BEGIN_SRC C++ -n :exports both :includes <iostream> :defines cout std::cout
+  #include <iostream>
+  using namespace std;
+
+  int main() {
+    cout << "Hello World!";
+    return 0;
+  }
+#+END_SRC
+
+* Click Run
+#+ATTR_HTML: :width 50%
+[[file:https://i.imgur.com/caYg2AY.png]]
+
+* {}
+- Curly Braces define scope of a function in the code
+- Right now that means that your code goes in the main function
+- All compound or control flow statements have curly braces
+    - Below the *{}* defines the main method
+#+BEGIN_SRC C++ -n
+  int main() {
+    // <==== HERE =====
+  }
+#+END_SRC
+
+* ;
+#+BEGIN_NOTES
+Make sure to note that they will see this error a lot and that it will go away with time.
+#+END_NOTES
+- A semicolon defines the end of a line of code
+- Think of it like a period in a sentence
+- All expression statements end in a semicolon
+    - declarations, assignments, function calls
+    - A missing semicolon results in an error message that can point to the wrong line
+#+BEGIN_SRC C++ -n
+    #include <iostream>
+    using namespace std;
+    int main() {
+      cout << "Hello" // <=====
+      cout << "World!";
+      return 0;
+    }
+#+END_SRC
+- Output
```

Comment:
Having the output there is fine, I'm being picky about you putting a bullet above it that says output. For consistent styling with the rest of the deck, I'd suggest removing it.